MicrosoftDX/nether
# Azure Data Lake Analytics scripts

### Issue: #PutYourIssueNumberHere


### Description:
Included usql queries for doing counts across geohashes. The result of running one such query is the following file in the Azure Data Lake Store (associated with the Azure Data Lake Analytics account):
/output/locations/{year}{month}{day}.csv: geoHash, count of points, median latitude, longitude of one point with given median latitude

3 usql queries:
- locations-dailyCount-medianCoords.usql: calculates the count of points per geoHash and calculates the median latitude per geoHash and a corresponding longitude (not necessarily the median longitude). On a daily basis.
- locations-monthlyCount-medianCoords.usql: same as above, only on a monthly basis. Hence, could have different coordinates as above.
- locations-yearlyCount-medianCoords.usql: same as above, only on an annual basis.

Abbreviations:
ADLA = Azure Data Lake Analytics
ADLS = Azure Data Lake Store

### Test:
1. Create an Azure Data Lake Analytics account: in [portal](https://docs.microsoft.com/en-us/azure/data-lake-analytics/data-lake-analytics-get-started-portal), through [PowerShell](https://docs.microsoft.com/en-us/azure/data-lake-analytics/data-lake-analytics-get-started-powershell) or [Azure CLI](https://docs.microsoft.com/en-us/azure/data-lake-analytics/data-lake-analytics-get-started-cli2)

2. EITHER in Visual Studio: Make sure you have the Data Lake Tools for Visual Studio (either already included in VS2017 or to be downloaded [here](http://aka.ms/adltoolsvs)).
OR in Visual Studio Code: Install the Visual Studio Code extension called [Azure Data Lake Tools](https://marketplace.visualstudio.com/items?itemName=usqlextpublisher.usql-vscode-ext) through the Visual Studio Code marketplace or by launching VS Code Quick Open (Ctrl+P) and executing the following command:
`ext install usql-vscode-ext

3. Open one of the usql queries under src/Nether.Analytics.ADLA, e.g. locations-dailyCount-medianCoords.usql

4. Submit the usql query by 
EITHER in Visual Studio: clicking "Submit" on the top left after specifying the previously created ADLA acount
OR in Visual Studio Code: 
a. compile the U-SQL script (`ADL: Compile Script`)
b. submit the U-SQL script (`ADL: Submit Job`)

5. Once the usql query has successfully run, you can open the output file in the corresponding Azure Data Lake Store, e.g. /output/locations/20170427.csv --> count of points per geoHash and its corresponding median latitude and longitude on April 27th 2017
You can either use the Azure Data Lake Explorer within Visual Studio or the Data Explorer in the Azure Portal to navigate to the output file.

A sample walkthrough of running U-SQL scripts in
- Visual Studio can be found [here](https://docs.microsoft.com/en-us/azure/data-lake-analytics/data-lake-analytics-data-lake-tools-get-started) with some screenshots.
- Visual Studio Code can be found [here](https://docs.microsoft.com/en-us/azure/data-lake-analytics/data-lake-analytics-data-lake-tools-for-vscode) with some screenshots.
